main.js

```diff
@@ -37,10 +37,10 @@ module.exports = (glob, options) => {
   // Add each created BrowserWindow to list of maintained items
   app.on('browser-window-created', (e, bw) => {
     browserWindows.push(bw)
-    let i = browserWindows.indexOf(bw)
 
     // Remove closed windows from list of maintained items
     bw.on('closed', function () {
+      let i = browserWindows.indexOf(bw) // Must use current index
```

Comment:
Wow! Does the index change during its life cycle? 😨

Reply:
The index of a window in the array changes when the splice function is called during the 'onclosed' event of an earlier window.

Try this:
Open one and two. Close one. Change files. Notice: Two does not reload.
After both are opened:  browserWindows[0] === one && browserWindows[1] === two
Closing one leaves it as: browserWindows[0] two

It only happens if browser windows are dynamically opened and closed by the app.

Reply:
That makes perfect sense. I should be ashamed of not seeing this! 😅